vNext/src/Orleans/Shims/AppDomain.cs

```diff
@@ -31,19 +34,31 @@ private AppDomain()
             //};
         }
 
-        public Assembly[] GetAssemblies()
+        private Lazy<Assembly[]> assembliesList = new Lazy<Assembly[]>(() =>
         {
-            // TODO: replace with IAssemblyCatalog or something like that.
-            Assembly[] assemblies =
+            string path = ".";
+            var assemblyFiles = Directory.EnumerateFiles(path, "*.dll");
```

Comment:
Not sure "." is the best way to get the directory which the current assembly resides in - wont this return the files in the working directory?

I find this to be reliable, particularly when dealing with things like shadow copying, etc:
```C#
var path = new Uri((this.GetType().Assembly).CodeBase).LocalPath;
```